src/code_execution/godbolt.rs

```diff
@@ -51,34 +56,188 @@ struct GodboltTool {
+                time_since_update,
+            );
+        }
+
+        needs_update
+    } else {
+        log::info!("godbolt targets haven't yet been updated, fetching them");
+
+        true
+    };
+
+    // If we should perform an update then do so
+    if needs_update {
+        let request = data
+            .http
+            .post(GODBOLT_TARGETS_URL)
+            .header(reqwest::header::ACCEPT, ACCEPT_JSON)
+            .build()?;
+
+        let mut targets: Vec<GodboltTarget> = data.http.execute(request).await?.json().await?;
+        log::info!("got {} godbolt targets", targets.len());
+
+        // Clean up the data we've gotten from the request
+        for target in &mut targets {
+            target.clean_request_data();
+        }
+
+        // Run the target updates within a transaction so that if things go wrong we don't
+        // end up with an empty targets list
+        conn.transaction::<_, (), Error>(|conn| {
+            Box::pin(async move {
+                // Remove all old values from the targets list, this is probably overly-cautious
+                // but it at least ensures that we never have incorrect targets within the db
+                sqlx::query!("DELETE FROM godbolt_targets")
+                    .execute(&mut *conn)
+                    .await?;
+
+                // Insert all of our newly fetched targets into the db
+                for target in targets {
+                    sqlx::query!(
+                        "INSERT INTO godbolt_targets (
+                            id,
+                            name,
+                            lang,
+                            compiler_type,
+                            semver,
+                            instruction_set
+                         )
+                         VALUES ($1, $2, $3, $4, $5, $6)",
+                        target.id,
+                        target.name,
+                        target.lang,
+                        target.compiler_type,
+                        target.semver,
+                        target.instruction_set,
+                    )
+                    .execute(&mut *conn)
+                    .await?;
+                }
+
+                // Get the current utc timestamp
+                let current_time = Utc::now().timestamp_nanos();
+
+                // Set the last godbolt update time to now
+                sqlx::query!(
+                    "INSERT INTO last_godbolt_update (last_update)
+                     VALUES ($1)
+                     ON CONFLICT(id) DO UPDATE SET last_update = $1
+                     WHERE id = 0",
+                    current_time,
+                )
+                .execute(&mut *conn)
+                .await?;
+
+                Ok(())
+            })
+        })
+        .await?;
+
+        log::info!("finished updating godbolt targets list");
+    }
+
+    Ok(conn)
+}
+
+async fn fetch_godbolt_targets<'a, C>(conn: C) -> Result<HashMap<String, String>, Error>
```

Comment:
Maybe this function could do update_and_fetch_godbolt_targets instead? That way, the boilerplate on lines 365-372 and 464-474 is hidden away.

The function would have to return instruction_set as well for that, perhaps via a `HashMap<String, (String, String)>`, `HashMap<String, GodboltTarget>` or `Vec<GodboltTarget>` (my favorite I think)

Unused fields from GodboltTarget could be dropped to reduce useless code